### PR TITLE
Xcode8 support +

### PR DESCRIPTION
**Pull request by slightair**
- Remove Quick & Nimble (use XCTest instead)
- Convert test code to Swift3
- Fix broken asynchronous test
- API changes for Swift3
- Update dependencies

### CODE REVIEW

File: Example/Podfile [hunk 1]
```diff
@@ -1,9 +1,6 @@
 source 'https://github.com/CocoaPods/Specs'
 use_frameworks!
```
Comment (giginet) [inline Example/Podfile:2]: Is this necessary after removing `Quick` and `Nimble`?

File: Example/Tests/PURLogStoreTest.swift [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        logStore = PURLogStore(databasePath: PURLogStoreTest.logStoreDBPath)
+        logStore.prepare()
+    }
+
+    override func tearDown() {
+        logStore.clearAll()
+    }
+
+    func assertLogStoreLogCount(pattern: String, output: PUROutput, expectedCount: Int, line: Int = #line) {
```
Comment (giginet) [inline Example/Tests/PURLogStoreTest.swift:28]: In my opinion, this utility's name is verbose.
I suppose Swift 3 style naming
`assertLogCount(for pattern: String, to output: PUROutput, shouldBe expectedCount: Int, on line: Int = #line)`

File: Example/Tests/PURLoggerStandardPluginTest.swift [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+        return configuration
+    }
+
+    func testChangeTagFilterPlugin() {
+        let loggerConfiguration = PURLoggerStandardPluginTest.loggerConfiguration
+        let logger = PURLogger(configuration: loggerConfiguration)
+        let testLogStorage = loggerConfiguration.logStorage
+
+        XCTAssertEqual(testLogStorage.description, "")
```
Comment (giginet) [inline Example/Tests/PURLoggerStandardPluginTest.swift:38]: You should use `String(describing: obj)` instead of calling `description` directly

File: Example/Tests/PURLoggerStandardPluginTest.swift [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        logger.post(["aaa": "3"], tag: "failure")
+        logger.post(["aaa": "4"], tag: "failure")
+        logger.post(["aaa": "5"], tag: "failure")
+
+        waitForExpectations(timeout: 1.0, handler: nil)
+        XCTAssertEqual(testLogStorage.description, "[error]")
+
+        expectation(forNotification: Notification.Name.PURBufferedOutputDidTryWriteChunk.rawValue, object: nil, handler: nil)
+        // scheduled after 2sec
+        waitForExpectations(timeout: 3.0, handler: nil)
```
Comment (giginet) [inline Example/Tests/PURLoggerStandardPluginTest.swift:224]: why these timeouts are different between its above comments?

File: Example/Tests/PURLoggerStandardPluginTest.swift [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+        logger.post(["aaa": "123"], tag: "filter.test")
+        logger.post(["bbb": "456", "ccc": "789"], tag: "filter.test")
+        logger.post(["ddd": "12345"], tag: "debug")
+        logger.post(["eee": "not filtered"], tag: "filter.testXXX")
+
+        XCTAssertEqual(testLogStorage.description, "[filter.testXXX|aaa:123][filter.testXXX|bbb:456,ccc:789][filter.testXXX|eee:not filtered]")
+
+        logger.logStore().clearAll()
+        logger.shutdown()
```
Comment (giginet) [inline Example/Tests/PURLoggerStandardPluginTest.swift:48]: These seems to be execute on every test cases.
Would you execute them on `tearDown`.

**Review by giginet — COMMENTED** [review]
Sorry for my late reply. 
I reviewed about 1 weeks ago however these had been pending 🙇

File: Example/Tests/PURLoggerStandardPluginTest.swift (hunk 5 above)
Comment (slightair) [reply Example/Tests/PURLoggerStandardPluginTest.swift:48]: I confused test methods behaviour. I fixed it. bcbff499e

File: Example/Tests/PURLoggerStandardPluginTest.swift (hunk 4 above)
Comment (slightair) [reply Example/Tests/PURLoggerStandardPluginTest.swift:224]: BufferedOutput schedules retry after 2s, 4s, 8s.
So test wait +1 seconds for every retries.

File: Pod/Classes/PURLogStore.m [hunk 6]
```diff
@@ -54,10 +51,15 @@ - (instancetype)initWithDatabasePath:(NSString *)databasePath
     self = [super init];
     if (self) {
         _databasePath = databasePath;
+        _databaseReadWriteCompletionQueue = dispatch_queue_create("PureeLogStoreReadWriteCompletion", NULL);
     }
     return self;
 }
 
+- (void)dealloc {
+    self.databaseReadWriteCompletionQueue = nil;
```
Comment (chuganzy) [inline Pod/Classes/PURLogStore.m:60]: [nit] This is unnecessary.

File: Pod/Classes/PURLogStore.m [hunk 7]
```diff
@@ -54,10 +51,15 @@ - (instancetype)initWithDatabasePath:(NSString *)databasePath
     self = [super init];
     if (self) {
         _databasePath = databasePath;
+        _databaseReadWriteCompletionQueue = dispatch_queue_create("PureeLogStoreReadWriteCompletion", NULL);
```
Comment (chuganzy) [inline Pod/Classes/PURLogStore.m:54]: Why did you change the completion queue to the subthread one?
If you change, you need to add some lock-logics to PURBufferedOutput class.

I think using the main-thread makes it simpler.

Comment (slightair) [reply Pod/Classes/PURLogStore.m:54]: You introduce async DB readwrite #6 for avoid UI blocking.
I think better that is some BufferedOutput processes running on the background too.

Do you think that DB read write completion block should running on the main-thread?

Comment (chuganzy) [reply Pod/Classes/PURLogStore.m:54]: Yes.

DB R/W process is too heavy, but compared to that, BufferedOutput's internal processes are not. Rather than that, using the main-thread makes it much more simpler. 
Anyway, currently `-(void)flush` is called from multiple thread and may sometimes cause NSRangeException. We should decide it as soon as possible.

Comment (slightair) [reply Pod/Classes/PURLogStore.m:54]: Please review
https://github.com/cookpad/puree-ios/pull/17

Comment (chuganzy) [reply Pod/Classes/PURLogStore.m:54]: Looks good 👍
